ams/server/src/main/java/com/netease/arctic/server/optimizing/OptimizingQueue.java

```diff
@@ -314,6 +320,11 @@ private void planTasks() {
           TableOptimizingProcess optimizingProcess = new TableOptimizingProcess(planner);
           LOG.info("{} after plan get {} tasks", tableRuntime.getTableIdentifier(),
               optimizingProcess.getTaskMap().size());
+          SelfOptimizingReport selfOptimizingReport =
```

Comment:
1. Only report `planner.isNecessary() == true`. I think `== false` should also be reported.
2. Do we need keep the cost value same with printed in the `com.netease.arctic.server.optimizing.plan.OptimizingPlanner#planTasks` ?
```
    LOG.info("{} finish plan, type = {}, get {} tasks, cost {} ns, {} ms", tableRuntime.getTableIdentifier(),
        getOptimizingType(), tasks.size(), endTime - startTime, (endTime - startTime) / 1_000_000);
```

Reply:
I agree report metric even when planner.isNecessary() == false